### ISSUE TITLE
Authentication as a GitHub App

### ISSUE THREAD

**Issue by webknjaz** [cla-assistant/cla-assistant#430]
Hi,

I've noticed that this project uses a dedicated user account. This approach has a number of limitations and is not very flexible.
It seems like a modern way of doing this is setting up a [GitHub App](https://developer.github.com/apps/) which is easily installable, has more flexible access control and a broader rate limiting boundaries.
I know that [Probot](https://probot.github.io/) supports it out of the box. So you could use it to avoid having to implement auth flows from scratch.

**Comment by KharitonOff** [comment]
@webknjaz yes, you are right. We also thought about it and would like to move CLA assistant to GitHub Apps... As soon we get more capacity to tackle this issue, we'll do it. But we are always open for any kind of support from our community too. So let's see how soon we"ll be able to do it.
Thank you very much for your advice, and in case you are willing to support us here a bit more we'd appreciate it very much 😉

**Comment by thojansen** [comment]
Our latest thoughts are that we would love to evaluate GitHub Actions and Probot and see how far we could get while eliminating the need of a centrally managed database to store contributors and rather move something like a `CONTRIBUTORS` file in the repository (currently we have quite a lot of self-hosted instance). Maybe this will end up in a spin-off with a reduced feature set, but a much simpler codebase.

**Comment by webknjaz** [comment]
Well, I don't have resources to contribute directly but I've got some knowledge (currently building a framework similar to Probot but in Python 3.7 — [octomachinery](https://octomachinery.dev/)).
So feel free to ping me should you have any questions about GitHub Apps type of integration :)

**Comment by jeffwilcox** [comment]
@thojansen interested to understand a little more about decentralizing the database of contributors... it would be a nice GitHub-native approach to things, but the signed Gists, who signed, company names, dates, etc., could be rather personal to some. Would that be somehow in the repo or where would that knowledge live?

**Comment by thojansen** [comment]
The idea here was really to tackle two points at once. If we would get rid of a central database, more companies would be able to use the hosted offering. Also switching to s.th. else would be easily possible if we could create a reasonable format.

For small projects, a simple `CONTRIBUTORS` file should do the job which would just list the GitHub handles. Maybe also add it to the readme as done by https://allcontributors.org/docs/en/overview.

For larger projects/ orgs. with custom fields etc. I would assume that we would need a private repo to store information. I also guess that you would like to have signatures from multiple repos at one single place. This would also allow you to manually edit information (our current upload/ download use cases).

Ideally, we would support both cases as not everyone has private repos ($$), but for larger projects with a reasonable size this should not be a blocker (also hosting your own db would be the alternative).

**Comment by webknjaz** [comment]
> just list the GitHub handles. 

The problem with this one is that whenever a user changes their handle it'll point to somebody else.

**Comment by thojansen** [comment]
More specifically, we could include the user id.

| id | login | company |
|---|-------|-----|
| 578543 | [webknjaz](https://github.com/webknjaz) | Ansible Core RedHatOfficial |

**Comment by webknjaz** [comment]
@thojansen user id sounds good. You could also include a github-generated email which includes that id.
It's visible at https://github.com/settings/emails and for me is `578543+webknjaz@users.noreply.github.com`. This is how GitHub addresses username changes now...

**Comment by webknjaz** [comment]
unstale

**Comment by jeffwilcox** [comment]
I still believe this is an important feature for the future of this system as a viable GitHub App... if only we had more time! Recommend not closing as stale.

**Comment by ibakshay** [comment]
Hi @jeffwilcox, Thank you for your comments.  Meanwhile,  we have developed a light version of CLA assistant using the `GitHub Actions` platform.  We have already rolled out an Alpha release and it will be great if you can check that solution out too 😄 . https://github.com/cla-assistant/github-action
